src/experimental/tools/download2go/api/downloader/initSegment.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright 2019 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IDBPDatabase } from "idb";
+import { EMPTY, merge, of, Subject } from "rxjs";
+import {
+ bufferCount,
+ catchError,
+ distinctUntilKeyChanged,
+ filter,
+ map,
+ mapTo,
+ mergeMap,
+ reduce,
+ tap,
+ timeout,
+} from "rxjs/operators";
+
+import { IContentProtection } from "../../../../../core/eme";
+import { SegmentPipelinesManager } from "../../../../../core/pipelines";
+import { IInitSettings } from "../../types";
+import { IndexDBError, SegmentConstuctionError } from "../../utils";
+import ContentManager from "../context/ContentsManager";
+import { ContentType } from "../context/types";
+import EMETransaction from "../drm/keySystems";
+import { manifestLoader } from "./manifest";
+import { handleSegmentPipelineFromContexts } from "./segment";
+import { ICustomSegment, IInitGroupedSegments, IInitSegment } from "./types";
+
+/**
+ * This function take care of downloading the init segment for VIDEO/AUDIO/TEXT
+ * buffer type.
+ * Then, he also find out the nextSegments we have to download.
+ *
+ * @param IInitSettings - Argument we need to start the download.
+ * @param IDBPDatabase - An instance of the IndexDB to store the init segment in base
```

Comment:
It's `IndexedDB`

Reply:
hahaha, right!, more I used it more I forgot her name.